doc/man3/EVP_PKEY_keygen.pod

```diff
@@ -103,14 +109,24 @@ a string parameter must be given to specify the name of the EC curve.
 If I<type> is C<X25519>, C<X448>, C<ED25519>, or C<ED448>
 no further parameter is needed.
 
+EVP_PKEY_generate_public_key() generates an B<EVP_PKEY> with public key
+components from I<private_key> that contains just private key components
+if the provider and the algorithm of the key supports the operation.
+If I<private_key> already contains the public key components the function
+duplicates I<private_key>.
```

Comment:
Why are we duplicating instead of up refing?

Reply:
We could in theory up-ref in case the keep_private is set, however as that seems to me a real corner case where in all the other cases the new key is a separate object it seems to me better to duplicate it in this case too.

Reply:
Yeah, ok. That makes sense.